src/sentinel.c

```diff
@@ -3774,35 +3847,72 @@ NULL
         };
         addReplyHelp(c, help);
     } else if (!strcasecmp(c->argv[1]->ptr,"masters")) {
-        /* SENTINEL MASTERS */
-        if (c->argc != 2) goto numargserr;
-        addReplyDictOfRedisInstances(c,sentinel.masters);
+        /* SENTINEL MASTERS [<param> <param>...]*/        
+        if (c->argc == 2)
+            addReplyDictOfRedisInstances(c,sentinel.masters,NULL);
+        else {
+            dict *params = dictCreate(&stateSetDictType);
+            for (int i = 2; i < c->argc; i++) {
+                sds param = sdsnew(c->argv[i]->ptr);
+                if (dictAdd(params, param, NULL) != DICT_OK)
+                    sdsfree(param);
+            }
+            addReplyDictOfRedisInstances(c,sentinel.masters,params);
+            releaseInfoSectionDict(params);
+        }        
```

Comment:
@moticless Hi Moti, follow you comments, I update my codes in the following:

1.  Place the static inline function after all typedef, just next to all other static functions.

2.  Rename the function to isAddParamToReply

3.  Refactor codes for masters, sentinels and slaves part because they share similar code structure

Please take a look, Thanks for your review.